lib/AArch64Base.ml

```diff
@@ -1708,6 +1708,81 @@ type 'k kinstruction =
   | I_STZ2G of reg * reg * 'k idx
   | I_LDG of reg * reg * 'k
   | I_UDF of 'k
+(* Pointer Authentication Code: Basic instruction only
+   See C3.1.10 for the complete list of instructions
+    - First part: Add PAC to a register or SP, and and return the value in a
+    general purpose register
+*)
+  (* see C6.2.266
+   * PACIA <Xd>, <Xn|Sp>
+   * PACIZA <Xd>, <X31>
+   * PACIASP <X30>, <Sp>
+   * PACIAZ <X30>, <X31>
+   * PACIA1716 <X17>, <X16>
+   *)
+  | I_PAC_IA of reg * reg     (* detination <- AddPACIA(source) *)
```

Comment:
Hi @RemyCiterin. It would be more rational do define a type `pac_key` (for instance):
```OCaml
type pac_key_t = IA | IB | DA | DB
```
And then to define a single "abstract" instruction:
```OCaml
..,
| I_PAC of pac_key_t * reg * reg
...

This technique yield shorter code in many situations. For instance printing
```OCaml

let pp_pack_key = function
| IA -> "IA" | IB -> "IB" | DA -> "DA" | DB -> "DB"

let do_pp+instruction = ....
...
| | I_PA (key,r1, r2) ->
            sprintf "PAC%S %s, %s"  (pp_pack_key key) (pp_reg r1) (pp_reg r2)
...
```
Similarily for other PAC instruction: better have a "key" argument than four instructions.